docs/docs/guide/project/dependencies.md

```diff
@@ -244,6 +244,55 @@ pod install
 > [!WARNING]
 > CocoaPods dependencies are not compatible with workflows like `build` or `test` that run `xcodebuild` right after generating the project. They are also incompatible with binary caching and selective testing since the fingerprinting logic doesn't account for the Pods dependencies.
 
+## Static or dynamic
+
+Frameworks and libraries can be linked either statically or dynamically, a choice that has significant implications for aspects like app size and boot time. Despite its importance, this decision is often made without much consideration. Tools like [Swift Package Manager](https://www.swift.org/documentation/package-manager/) and [Mergeable Libraries](https://developer.apple.com/documentation/xcode/configuring-your-project-to-use-mergeable-libraries) automatically make this choice during build-time, which, while convenient, adds complexity to the build process and introduces potential sources of non-determinism. This can cause some Xcode features, such as SwiftUI previews, to become unreliable.
```

Comment:
could we be more explicit about what causes non-determinism? Apple's solution to this problem is intended to be mergable libraries, so it'd be great to be explicit about what the current blockers/challenges are there without telling folks to completely avoid the feature. I feel we need some clearly documented risks/issues if we're recommending *against* Apple solutions.

Reply:
I'm still on the fence about whether we should also automatically use `.static` or `.dynamic`. It's causing a lot of friction for users and while I do think it's important for users to understand the difference, it's also daunting to start using tuist and having bunch of duplicate symbols warnings that you wouldn't get with SPM.

I think we should consider respecting the `.automatic` product type while still giving the possibility to users to make the product type explicit (which SPM doesn't allow)

Reply:
> I'm still on the fence about whether we should also automatically use .static or .dynamic. It's causing a lot of friction for users and while I do think it's important for users to understand the difference, it's also daunting to start using tuist and having bunch of duplicate symbols warnings that you wouldn't get with SPM.

But this convenience over explicitness was one of the motivators behind building Tuist. While we could conceptually compress some complexity that users don't need to be thinking about, like the build phases or settings necessary to link A with B, we can't be making decisions about how things should be linked, even if that means users need to read a bit more. The implications of these decisions are deep, starting from potentially increasing users' binary sizes if we get it wrong, or causing apps to blow at runtime after they are released.

Tuist should be educational in this area, and provide the best resources possible for developers to understand what is linking, why it is so important, and how to make the decisions themselves. I wouldn't prevent users from having to read here.